### PR TITLE
Display error messages

### PR DESCRIPTION
**Pull request by petro97**
Description:

Linting:

- [x] No linting errors

Tests:

- [ ] E2E tests
- [x] Manual tests
- [ ] Accessibility tests (no new react-axe errors in console)

Documentation:

- [ ] Requires documentation updates
- [ ] Requires Storybook component updates

### CODE REVIEW

File: src/APIErrorNotification.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  const { error, removeError } = useContext(APIErrorContext);
+
+  const handleCloseModal = () => {
+    removeError()
+  };
+
+  return (
+    <div className="notification">
+      {!!error && error.length > 0 && error.map((el: IErrorMessage) => (
+        <div className="notification__body" key={`${el.detail}_${Math.random().toString(36)}`}>
```
Comment (shaunmaharaj) [inline src/APIErrorNotification.tsx:27]: I look forward to when we can get error message keys from the API to populate this 😄

File: src/App.tsx [hunk 2]
```diff
@@ -17,33 +19,36 @@ const App: React.FC = () => {
...
-            </div>
-          </footer>
-          <aside className="app__compareoverlay">
-            <CompareOverlay />
-          </aside>
-        </div>
-        </InstantSearch>
-      </AppStateProvider>
-    </Router>
+    <APIErrorProvider>
```
Comment (shaunmaharaj) [inline src/App.tsx:22]: Any concern with the error provider being outside our router? Could it impact our routing? Not sure, just generally asking.

File: src/Promotion.tsx [hunk 3]
```diff
@@ -34,6 +36,7 @@ export const Promotion: React.FC<PromotionProps> = (props) => {
         })
         .catch(error => {
           console.error(error);
+          addError(error.errors);
```
Comment (shaunmaharaj) [inline src/Promotion.tsx:39]: Do we still need to call setErrors if we're displaying our own toast?

File: src/app-state.ts [hunk 4]
```diff
@@ -1,8 +1,14 @@
 import { useState, useEffect } from 'react';
 import constate from 'constate';
 import * as moltin from '@moltin/sdk';
-import { getCustomer, getAddresses, getAllOrders, loadCategoryTree, getCartItems } from './service';
-import * as service from './service';
+import {
+  getCustomer,
```
Comment (shaunmaharaj) [inline src/app-state.ts:5]: Thanks for cleaning this up!

**Review by shaunmaharaj — COMMENTED** [review]
Thank you for this Petro! I left a couple comments to clarify. We can merge once resolved

File: src/Promotion.tsx (hunk 3 above)
Comment (petro97) [reply src/Promotion.tsx:39]: @shaunmaharaj,
I think we should display our own toast.
That's actually what I told before. Should we use the state of a component or our own toast for displaying messages (the same is in address form, login form, and registration form)?

File: src/App.tsx (hunk 2 above)
Comment (petro97) [reply src/App.tsx:22]: It shouldn't impact our routing. But I can move it inside the router

**Review by shaunmaharaj — APPROVED** [review]

**Pull request merged**